lib/portage/checksum.py

```diff
@@ -207,101 +141,26 @@ def hexdigest(self):
-            )
-    except ImportError:
-        pass
-
-if "BLAKE2S" not in hashfunc_map:
-    try:
-        from Crypto.Hash import BLAKE2s
-
-        blake2shash_ = getattr(BLAKE2s, "new", None)
-        if blake2shash_ is not None:
-            _generate_hash_function(
-                "BLAKE2S",
-                functools.partial(blake2shash_, digest_bytes=32),
-                origin="pycrypto",
-            )
-    except ImportError:
-        pass
-
-if "SHA3_256" not in hashfunc_map:
-    try:
-        from Crypto.Hash import SHA3_256
-
-        sha3_256hash_ = getattr(SHA3_256, "new", None)
-        if sha3_256hash_ is not None:
-            _generate_hash_function("SHA3_256", sha3_256hash_, origin="pycrypto")
-    except ImportError:
-        pass
-
-if "SHA3_512" not in hashfunc_map:
-    try:
-        from Crypto.Hash import SHA3_512
-
-        sha3_512hash_ = getattr(SHA3_512, "new", None)
-        if sha3_512hash_ is not None:
-            _generate_hash_function("SHA3_512", sha3_512hash_, origin="pycrypto")
-    except ImportError:
-        pass
-
-
-# Try to use mhash if available
-# mhash causes GIL presently, so it gets less priority than hashlib and
-# pycrypto. However, it might be the only accelerated implementation of
-# WHIRLPOOL available.
-if "RMD160" not in hashfunc_map or "WHIRLPOOL" not in hashfunc_map:
-    try:
-        import mhash
-
-        for local_name, hash_name in (
-            ("RMD160", "RIPEMD160"),
-            ("WHIRLPOOL", "WHIRLPOOL"),
-        ):
-            if local_name not in hashfunc_map and hasattr(mhash, f"MHASH_{hash_name}"):
-                _generate_hash_function(
-                    local_name,
-                    functools.partial(
-                        mhash.MHASH, getattr(mhash, f"MHASH_{hash_name}")
-                    ),
-                    origin="mhash",
-                )
-    except ImportError:
-        pass
-
-
-# Support pygost as fallback streebog provider
-# It's mostly provided as a reference implementation; it's pure Python,
-# slow and reads all data to memory (i.e. doesn't hash on update()...)
-if "STREEBOG256" not in hashfunc_map or "STREEBOG512" not in hashfunc_map:
-    try:
-        import pygost.gost34112012
-
-        _generate_hash_function(
-            "STREEBOG256",
-            functools.partial(pygost.gost34112012.GOST34112012, digest_size=32),
-            origin="pygost",
-        )
-        _generate_hash_function(
-            "STREEBOG512",
-            functools.partial(pygost.gost34112012.GOST34112012, digest_size=64),
-            origin="pygost",
-        )
-    except ImportError:
-        pass
+        # Try to use mhash if available
+        # mhash causes GIL presently, so it gets less priority than hashlib and
+        # pycrypto. However, it might be the only accelerated implementation of
+        # WHIRLPOOL available.
+        try:
+            import mhash
+
+            for local_name, hash_name in (("RMD160", "RIPEMD160"),):
```

Comment:
I'd say you can drop RIPEMD as well.

Reply:
From mhash (leaving pycrypto* as the provider) or entirely? I'm trying to avoid dropping things entirely just yet, but want to do that later on.

Reply:
I was thinking of mhash since it's its only use.

Reply:
Hmm, though I guess after this commit it's also only use of pycrypto.